boards: frdm_mcxc444: Add counters support

NeilChen93 opened the pull request:
mcxc444 also has pit and rtc counters, add the counters to board documentation, and enable it explicitly in board dts. Set rtc clock to 32 kHz oscillator.

mmahadevan108 commented:
@NeilChen93, please fix the merge conflict. Also kindly provide details on how these counters were tested.

NeilChen93 committed:
boards: frdm_mcxc444: Add counters support

mcxc444 also has pit and rtc counters, add the counters to
board documentation, and enable it explicitly in board dts.
Set rtc clock to 32 kHz oscillator.

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

mmahadevan108 commented on doc/releases/release-notes-4.1.rst:
```diff
@@ -93,6 +93,8 @@ Boards & SoC Support
 
 * Made these board changes:
 
+  * Enabled RTC, PIT on NXP ``frdm_mcxc444``
```
@NeilChen93 , apologies for the back and forth on this. Upon further discussion with the 4.1 release managers it was decided that we do not want to include such changes in the release notes. Kindly delete this addition.

dleach02 replied:
I dropped the commit for you @NeilChen93

NeilChen93 replied:
Thanks
